utils/writeable/src/lib.rs

```diff
@@ -104,6 +104,13 @@ pub mod adapters {
             self.0.writeable_length_hint()
         }
     }
+
+    impl<T: TryWriteable> fmt::Display for LossyWrap<T> {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            let _ = self.0.try_write_to(f)?;
+            Ok(())
```

Comment:
can you do `self.0.try_write_to(f)` here? The Result seems to match, no?

Reply:
`self.0.try_write_to(f)` returns: `Result<Result<(), Self::Error>, fmt::Error>`

I use the `?` to return the `fmt::Error` and the `let _ =` to discard the `Self::Error`. I think `let _ =` is a conventional way of explicitly discarding a Result.

Another way to write this would be: `self.0.try_write_to(f).map(|_| ())`

Reply:
ah, I see, no, it's ok. The nested Result tricked me. I'm wondering if having it flattened in `try_write_to` would be a better choice (with a bound that Self::Error has From<fmt::Error>), but that's not worth much effort right now :)

Reply:
I have a record of your hesitation to the double result in https://github.com/unicode-org/icu4x/pull/4772#issuecomment-2037828466

I still stand by the double result being better than the alternatives but it's maybe worth discussing again now that there is more context

Reply:
The double result was a very conscious choice, because if `Self::Error` would have to have an `FmtError` case, `.to_string()` could produce `FmtError`s. The `Result<_, fmt::Error>` can be safely unwrapped when writing to a `String`, but if it's a `Result<_, Self::Error>`, that other error would still have `FmtError` in it even when producing a `String`. Datetime used to do this and this was the motivation for `TryWriteable`.